mailbox/api/src/test/java/org/apache/james/mailbox/events/ErrorHandlingContract.java

```diff
@@ -154,4 +196,45 @@ default void retriesBackOffShouldDelayByExponentialGrowth() throws Exception {
                 .isAfterOrEqualTo(timeElapsed.get(2).plusMillis(minThirdDelayAfter));
         });
     }
+
+    @Test
+    default void deadLetterShouldNotStoreWhenFailsLessThanMaxRetries() {
+        EventCollector eventCollector = eventCollector();
+
+        doThrow(new RuntimeException())
+            .doThrow(new RuntimeException())
+            .doCallRealMethod()
+            .when(eventCollector).event(EVENT);
+
+        eventBus().register(eventCollector, new EventBusTestFixture.GroupA());
+        eventBus().dispatch(EVENT, NO_KEYS).block();
+
+        WAIT_CONDITION
+            .until(() -> assertThat(eventCollector.getEvents()).hasSize(1));
+
+        assertThat(deadLetter().groupsWithFailedEvents().toIterable())
+            .isEmpty();
+    }
+
+    @Test
+    default void deadLetterShouldStoreWhenFailsGreaterThanMaxRetries() throws Exception {
+        EventCollector eventCollector = eventCollector();
+
+        doThrow(new RuntimeException())
+            .doThrow(new RuntimeException())
+            .doThrow(new RuntimeException())
+            .doThrow(new RuntimeException())
+            .doCallRealMethod()
+            .when(eventCollector).event(EVENT);
+
+        eventBus().register(eventCollector, GROUP_A);
+        eventBus().dispatch(EVENT, NO_KEYS).block();
+
+        TimeUnit.SECONDS.sleep(1);
```

Comment:
Can't we await `on event is stored in EventDeadLetters` and then check `there is no event suuccessfully processed` so that we don't need the wait?